router/router.go

```diff
@@ -604,12 +638,33 @@ func (worker *workerT) handleWorkerDestinationJobs(ctx context.Context) {
 							//response transform start
 							if worker.rt.responseTransform {
 								dResponse := integrations.DeliveryResponseT{
-									Status: int64(respStatusCode),
-									Body:   respBodyTemp,
+									Status:  int64(respStatusCode),
+									Body:    respBodyTemp,
+									Payload: val,
 								}
+								// destination response status code
+								destRespStCd := respStatusCode
+								// destination response body
+								destRespBody := respBodyTemp
+
 								rtl_time := time.Now()
 								respStatusCode, respBodyTemp = worker.rt.transformer.ResponseTransform(ctx, dResponse, worker.rt.destName)
 								worker.routerResponseTransformStat.SendTiming(time.Since(rtl_time))
+								//Handle OAuth Destinations' Response
+								authType := router_utils.GetAuthType(destinationJob.Destination)
+								if router_utils.IsNotEmptyString(authType) && authType == "OAuth" {
+									pkgLogger.Debugf(`Sending for OAuth destination`)
+									respStatusCode, respBodyTemp = worker.rt.HandleOAuthDestResponse(&HandleDestOAuthRespParamsT{
+										ctx:            ctx,
+										val:            val,
+										destinationJob: destinationJob,
+										workerId:       worker.workerID,
+										destRespStCd:   destRespStCd,
+										destResBody:    destRespBody,
+										trRespStCd:     respStatusCode,
+										trRespBody:     respBodyTemp,
+									})
+								}
 							}
 							// response transform end
 							if isSuccessStatus(respStatusCode) {
```

Comment:
We have the option of not saving the destination response in case of success like this, did we remove that?
`	if isSuccessStatus(respStatusCode) && !worker.rt.saveDestinationResponseOverride {
					if saveDestinationResponse {
						if !getRouterConfigBool("saveDestinationResponse", worker.rt.destName, true) {
							respBody = ""
						}
					} else {
						respBody = ""
					}
				}`

Reply:
I am not sure about removing, `router.go` I've used the source code implemented by you in major parts like this.

Reply:
Also if you see Line numbers 694 - 702 you'll find what you were asking for. Can you confirm ?

Reply:
yeah found it was looking at wrong place no issues